lib/dao.js

```diff
@@ -468,6 +468,18 @@ DataAccessObject.updateOrCreate = DataAccessObject.upsert = function upsert(data
   var Model = this;
   var connector = Model.getConnector();
 
+  if (Array.isArray(data)) {
+    var isBulkCreate = data.every(function(item){
```

Comment:
Yes, that's correct. We are trying to preserve as much of the existing behavior as possible, while dropping support for cases where the function may crash.

When you pass an array of instances without it to create, we will still call `create`, as that's the current behavior and it's likely to pass (assuming your model data is valid).

When you pass at least on instance with id set, create would either pass or fail depending on whether there is another instance with the same id already saved. In this case, we will start reporting a useful error - that's the intention.